contracts/coordinator/src/query.rs

```diff
@@ -18,3 +20,52 @@ pub fn get_active_worker_set(deps: Deps, chain_name: ChainName) -> StdResult<Opt
 
     Ok(active_worker_set)
 }
+
+pub fn get_next_worker_set(deps: Deps, chain_name: ChainName) -> StdResult<Option<WorkerSet>> {
+    let prover_address = provers(deps, chain_name).unwrap();
+    let next_worker_set =
+        NEXT_WORKERSET_FOR_PROVER.may_load(deps.storage, prover_address.clone())?;
+
+    Ok(next_worker_set)
+}
+
+fn is_worker_in_worker_set(
+    deps: Deps,
+    chain_name: &ChainName,
+    worker_address: &Addr,
+) -> StdResult<bool> {
+    if let Ok(Some(worker_set)) = get_active_worker_set(deps, chain_name.clone()) {
+        if worker_set
+            .signers
+            .values()
+            .any(|signer| signer.address == *worker_address)
+        {
+            return Ok(true);
+        }
+    }
+
+    if let Ok(Some(worker_set)) = get_next_worker_set(deps, chain_name.clone()) {
+        if worker_set
+            .signers
+            .values()
+            .any(|signer| signer.address == *worker_address)
+        {
+            return Ok(true);
+        }
+    }
+
+    Ok(false)
+}
```

Comment:
you could probably simplify this and write it as just one function:
get the prover address
get the active workerset
get the next workerset
join the sets
check if the worker is in the joined set

Reply:
I have some unit tests that rely on the individual functions (get active and get next) so we can't remove the functions.